src/Match/GenericCase.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Munus\Match;
+
+use Munus\Match\GenericCase\GenericCaseCallable;
+use Munus\Match\GenericCase\GenericCaseStatic;
+
+/**
+ * @template T
+ */
+abstract class GenericCase implements MatchCase
+{
+    /**
+     * @var T|Predicate<T>
+     */
+    protected $value;
+
+    /**
+     * @psalm-template U
+     *
+     * @param T|Predicate<T> $value
+     * @param callable(T):U  $callable
+     */
+    public static function of($value, callable $callable): GenericCase
+    {
+        return new GenericCaseCallable($value, $callable);
+    }
+
+    /**
+     * @psalm-template U
+     *
+     * @param T|Predicate<T> $value
+     * @param U              $other
+     */
+    public static function ofStatic($value, $other): GenericCase
+    {
+        return new GenericCaseStatic($value, $other);
+    }
+
```

Comment:
Those `of`, and `ofStatic` are not the most elegant names. I thought about `of` for static values, and `call` for callable. The other solution may be to use two types of parameters and leave only `of`, but if someone will want to return the `callable` without executing, they will have to use the `callable` that returns `callable`, which is not very elegant.
Maybe someone knows a better solution, or likes one of my propositions above?

Reply:
I changed it to `of` and `call`, but it's still a request for comments 😄

Reply:
`of` and `call` looks ok :+1: